tests/UpdatableTimer/UpdatableTimerWorkflowTests.cs

```diff
@@ -0,0 +1,106 @@
+﻿namespace TemporalioSamples.Tests.UpdatableTimer;
+
+using Temporalio.Testing;
+using Temporalio.Worker;
+using TemporalioSamples.UpdatableTimer;
+using Xunit;
+using Xunit.Abstractions;
+
+public class UpdatableTimerWorkflowTests : TestBase
+{
+    public UpdatableTimerWorkflowTests(ITestOutputHelper output)
+        : base(output)
+    {
+    }
+
+    [Fact]
+    public async Task SimpleRun_Succeeds()
+    {
+        await using var env = await WorkflowEnvironment.StartLocalAsync();
+        using var worker = new TemporalWorker(env.Client, new TemporalWorkerOptions("my-task-queue").AddWorkflow<MyWorkflow>());
+        var wakeUpTime = DateTimeOffset.UtcNow;
+        await worker.ExecuteAsync(async () =>
+        {
+            var handle = await env.Client.StartWorkflowAsync(
+                (MyWorkflow wf) => wf.RunAsync(wakeUpTime),
+                new(id: $"workflow-{Guid.NewGuid()}", taskQueue: worker.Options.TaskQueue!));
+            await handle.GetResultAsync();
+        });
+    }
+
+    [Fact]
+    public async Task WakeUpInThePast_Succeeds()
+    {
+        await using var env = await WorkflowEnvironment.StartLocalAsync();
+        using var worker = new TemporalWorker(env.Client, new TemporalWorkerOptions("my-task-queue").AddWorkflow<MyWorkflow>());
+        var wakeUpTime = DateTimeOffset.UtcNow.AddSeconds(-10);
+        await worker.ExecuteAsync(async () =>
+        {
+            var handle = await env.Client.StartWorkflowAsync(
+                (MyWorkflow wf) => wf.RunAsync(wakeUpTime),
+                new(id: $"workflow-{Guid.NewGuid()}", taskQueue: worker.Options.TaskQueue!));
+            await handle.GetResultAsync();
+        });
+    }
+
+    [Fact]
+    public async Task WakeUpAfter30Days_Succeeds()
+    {
+        await using var env = await WorkflowEnvironment.StartTimeSkippingAsync();
+        using var worker = new TemporalWorker(env.Client, new TemporalWorkerOptions("my-task-queue").AddWorkflow<MyWorkflow>());
+        var wakeUpTime = DateTimeOffset.UtcNow.AddDays(30);
+
+        var startTime = await env.GetCurrentTimeAsync();
+        await worker.ExecuteAsync(async () =>
+        {
+            var handle = await env.Client.StartWorkflowAsync(
+                (MyWorkflow wf) => wf.RunAsync(wakeUpTime),
+                new(id: $"workflow-{Guid.NewGuid()}", taskQueue: worker.Options.TaskQueue!));
+
+            await AssertMore.EventuallyAsync(async () =>
```

Comment:
Would move this to after `GetResultAsync` and not have it be an `EventuallyAsync` but rather just check the history inline